src/plugins/charts/public/static/components/color_picker.tsx

```diff
@@ -74,54 +81,84 @@ export const legendColors: string[] = [
 ];
 
 interface ColorPickerProps {
-  id?: string;
+  /**
+   * Label that characterizes the color that is going to change
+   */
   label: string | number | null;
+  /**
+   * Callback on the color change
+   */
   onChange: (color: string | null, event: BaseSyntheticEvent) => void;
+  /**
+   * Initial color.
+   */
   color: string;
+  /**
+   * Defines if the compatibility (legacy) or eui palette is going to be used. Defauls to true.
+   */
+  useLegacyColors?: boolean;
+  /**
+   * Defines if the default color is overwritten. Defaults to true.
+   */
+  colorIsOverwritten?: boolean;
 }
 
-export const ColorPicker = ({ onChange, color: selectedColor, id, label }: ColorPickerProps) => (
-  <div className="visColorPicker">
-    <span id={`${id}ColorPickerDesc`} className="euiScreenReaderOnly">
-      <FormattedMessage
-        id="charts.colorPicker.setColor.screenReaderDescription"
-        defaultMessage="Set color for value {legendDataLabel}"
-        values={{ legendDataLabel: label }}
-      />
-    </span>
-    <div className="visColorPicker__value" role="listbox">
-      {legendColors.map((color) => (
-        <EuiIcon
-          role="option"
-          tabIndex={0}
-          type="dot"
-          size="l"
-          color={selectedColor}
-          key={color}
-          aria-label={color}
-          aria-describedby={`${id}ColorPickerDesc`}
-          aria-selected={color === selectedColor}
-          onClick={(e) => onChange(color, e)}
-          onKeyPress={(e) => onChange(color, e)}
-          className={classNames('visColorPicker__valueDot', {
-            // eslint-disable-next-line @typescript-eslint/naming-convention
-            'visColorPicker__valueDot-isSelected': color === selectedColor,
-          })}
-          style={{ color }}
-          data-test-subj={`visColorPickerColor-${color}`}
-        />
-      ))}
+const euiColors = euiPaletteColorBlind({ rotations: 4, order: 'group' });
+
+export const ColorPicker = ({
+  onChange,
+  color: selectedColor,
+  label,
+  useLegacyColors = true,
+  colorIsOverwritten = true,
+}: ColorPickerProps) => {
+  const legendColors = useLegacyColors ? legacyColors : euiColors;
+
+  return (
+    <div className="visColorPicker">
+      <EuiFlexGroup wrap={true} gutterSize="none" className="visColorPicker__value">
```

Comment:
A `<legend>` needs to be the first child of of a `<fieldset>` for it to setup a proper association. I'm guessing you need this `FlexGroup` around all of the `label` elements though so I'd put this wrapper just around the `.map()` you have below. (Don't worry about `<EuiScreenReaderOnly>` - it doesn't actually render an element in the DOM.)

```suggestion
      <fieldset>
```